### PR TITLE
Replace usage of MLLib and RDDs with ML and DFs

### PR DESCRIPTION
**Pull request by amCap1712**
Spark ML is the newer machine library of spark and is based on dataframes instead of RDDs. This PR replaces usage of Spark MLLib the older version with the newer one. This also helps with replacing some hand rolled code with Spark built-in apis.

### CODE REVIEW

File: listenbrainz_spark/recommendations/recording/train_models.py [hunk 1]
```diff
@@ -92,7 +58,7 @@ def preprocess_data(transformed_listencounts_df):
             test_data (rdd): Used for testing.
     """
     logger.info('Splitting dataframe...')
-    training_data, validation_data, test_data = transformed_listencounts_df.rdd.map(parse_dataset).randomSplit([4, 1, 1], 45)
+    training_data, validation_data, test_data = transformed_listencounts_df.randomSplit([4.0, 1.0, 1.0], 45)
```
Comment (mayhem) [inline listenbrainz_spark/recommendations/recording/train_models.py:61]: at least documenting these constants would be good.

**Review by mayhem — COMMENTED** [review]
So far so good! I really appreciate how this simplifies things AND brings us to use the modern APIs, which match up when you search the docs. Big thumbs up!

File: listenbrainz_spark/recommendations/recording/train_models.py (hunk 1 above)
Comment (amCap1712) [reply listenbrainz_spark/recommendations/recording/train_models.py:63]: Done.

**Review by mayhem — APPROVED** [review]
LGTM

**Pull request merged**